### PR TITLE
add .github/workflows/needs-rebase.yml to auto-tag PRs needing rebase

### PR DESCRIPTION
**Pull request by timotheecour**
## links
https://github.community/t/run-actions-on-pull-requests-with-merge-conflicts/17104
https://github.com/eps1lon/actions-label-merge-conflict
https://github.com/isaacs/github/issues/224
(another similar action was https://github.com/mschilde/auto-label-merge-conflicts)

## example
see conclicts label in this other project: https://github.com/OpenShot/libopenshot/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+label%3Aconflicts

image

### CODE REVIEW

**Comment by juancarlospaco** [comment]
```yml
if: !contains(github.event.head_commit.message, '[skip ci]')
```
?

File: .github/workflows/needs-rebase.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if PR needs rebase
+        uses: eps1lon/actions-label-merge-conflict@releases/2.x
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          dirtyLabel: "needs-rebase"
+          removeOnDirtyLabel: "no conflicts"
+          commentOnDirty: "conflicts detected, rebase needed"
+          commentOnClean: "conflicts have been resolved"
```
Comment (Clyybber) [inline .github/workflows/needs-rebase.yml]: I think this would get a bit spammy, just adding/removing the label without a comment is better.

Comment (timotheecour) [reply .github/workflows/needs-rebase.yml]: done; the point is to trigger a notification to PR author though for quicker turnaround; if the auto-label-change doesn't trigger a notification, we may need to at least re-enable `commentOnDirty`

**Comment by timotheecour** [comment]
PTAL 
> if: !contains(github.event.head_commit.message, '[skip ci]')

I'm not sure about that; this pipeline should run very fast (unlike rest of CI which has to build nim, run tests etc). Let's measure after this PR is merged, and if I'm wrong about that we can add your suggestion

**Review by ringabout — COMMENTED** [review]
Why we need this? I don't see many benefits?

I need people to review my PRs so that I can make more progresses rather than a bot reminds me of resolving conflicts.

**Pull request closed without merging**